src/Language/PureScript/AST/Declarations.hs

```diff
@@ -743,6 +743,9 @@ data Expr
   --
   | App Expr Expr
   -- |
+  -- Hint that an expression is unused
+  | Unused Expr
```

Comment:
Might be worth explaining why we need the extra `Expr` at all here (i.e. because we need to track unsolved constraints, I assume).

Reply:
Yes that's right, will update, thanks!